github/handlers/GithubPRlinkHandler.ts

```diff
@@ -1,62 +1,45 @@
-    try {
-        const githubPRLinkRegex = /\bhttps?:\/\/github\.com\/\S+\/pull\/\d+\b/;
-        const text = message.text!;
-        const prLinkMatch = text.match(githubPRLinkRegex);
-        const prLink = prLinkMatch?.[0];
-        const githubLinkPartsRegex = /(?:https?:\/\/github\.com\/)(\S+)\/(\S+)\/pull\/(\d+)/;
-        const linkPartsMatch = prLink?.match(githubLinkPartsRegex);
-        const username = linkPartsMatch?.[1];
-        const repositoryName = linkPartsMatch?.[2];
-        const pullNumber = linkPartsMatch?.[3];
-
-        if (!username || !repositoryName || !pullNumber) {
-            throw new Error("Invalid GitHub PR link");
-        }
-
-        const messageBuilder = await modify.getCreator().startMessage()
-            .setRoom(message.room)
-            .setSender(message.sender)
-            .setGroupable(true);
-
-        const block = modify.getCreator().getBlockBuilder();
-
-        block.addActionsBlock({
-            blockId: "githubdata",
-            elements: [
-                block.newButtonElement({
-                    actionId: ModalsEnum.MERGE_PULL_REQUEST_ACTION,
-                    text: block.newPlainTextObject("Merge"),
-                    value: `${username}/${repositoryName} ${pullNumber}`,
-                    style: ButtonStyle.PRIMARY
-                }),
-                block.newButtonElement({
-                    actionId: ModalsEnum.PR_COMMENT_LIST_ACTION,
-                    text: block.newPlainTextObject("Comment"),
-                    value: `${username}/${repositoryName} ${pullNumber}`,
-                    style: ButtonStyle.PRIMARY
-                }),
-                block.newButtonElement({
-                    actionId: ModalsEnum.APPROVE_PULL_REQUEST_ACTION,
-                    text: block.newPlainTextObject("Approve"),
-                    value: `${username}/${repositoryName} ${pullNumber}`,
-                    style: ButtonStyle.PRIMARY
-                })
-            ]
-        })
+import { IHttp, IMessageBuilder, IMessageExtender, IModify, IPersistence, IRead } from "@rocket.chat/apps-engine/definition/accessors";
+import { IMessage, IMessageAttachment, MessageActionButtonsAlignment, MessageActionType } from "@rocket.chat/apps-engine/definition/messages";
+import { IRoom } from "@rocket.chat/apps-engine/definition/rooms";
+
+export async function handleGithubPRLinks(
+    message: IMessage,
+    read: IRead,
+    http: IHttp,
+    user: IUser,
+    room: IRoom,
+    extend: IMessageExtender
+) {
+    const githubPRLinkRegex = /https?:\/\/github\.com\/(\S+)\/(\S+)\/pull\/(\d+)/g;
+    const text = message.text!;
+    let prLinkMatches: RegExpExecArray | null;
+    const matches: RegExpExecArray[] = [];
+
+    while ((prLinkMatches = githubPRLinkRegex.exec(text)) !== null) {
+        matches.push(prLinkMatches);
+    }
```

Comment:
This while loop is not making sense

Reply:
We are using this loop to get all the possible PR links from the message.

This while loop iterates through the matches of the regular expression in the message text. It uses `exec()` to find the next match in the text. As long as there is a match `(!== null)`, it adds the match to the matches array.